### PR TITLE
Document the ability to register custom elements

### PR DESCRIPTION
**Pull request by mansakondo**
This PR adds documentation to allow users to register custom elements.

### CODE REVIEW

File: docs/pages/views.rb [hunk 1]
```diff
@@ -130,6 +130,34 @@ def template(&content)
...
+				render Example.new do |e|
+					e.tab "hello.rb", <<~RUBY
+            class Hello < Phlex::View
+              CUSTOM_ELEMENTS = {
+                hello: "hello"
+              }.freeze
+
+              CUSTOM_ELEMENTS.each do |method_name, tag|
+                register_element(method_name, tag)
+              end
```
Comment (joeldrapper) [inline docs/pages/views.rb:149]: I think the golden path should be to just do this:

```suggestion
              register_element :element_name
```

File: docs/pages/views.rb [hunk 2]
```diff
@@ -130,6 +130,34 @@ def template(&content)
 					end
 					```
 				MD
+
+				render Markdown.new(<<~MD)
+					## Registering custom elements
+
+					Here's how you can register custom elements.
```
Comment (joeldrapper) [inline docs/pages/views.rb:137]:
```suggestion
					Here's how you can register custom elements. The custom element will only be available in the view where it is registered and subclasses of that view.
```

File: docs/pages/views.rb (hunk 1 above)
Comment (mansakondo) [reply docs/pages/views.rb:149]: Indeed, that's way cleaner.

File: docs/pages/views.rb [hunk 3]
```diff
@@ -130,6 +130,28 @@ def template(&content)
...
+				render Markdown.new(<<~MD)
+					## Registering custom elements
+
+					You can register custom elements with the `register_element` macro. The custom element will only be available in the view where it is registered and subclasses of that view.
+				MD
+
+				render Example.new do |e|
+					e.tab "hello.rb", <<~RUBY
+            class Hello < Phlex::View
+              register_element :hello
```
Comment (joeldrapper) [inline docs/pages/views.rb:143]: HTML custom elements must contain a dash, so let's include that in our example.

```suggestion
              register_element :trix_editor
```

File: docs/pages/views.rb [hunk 4]
```diff
@@ -130,6 +130,28 @@ def template(&content)
...
+
+              def template
+                hello do
+                  "World!"
+                end
```
Comment (joeldrapper) [inline docs/pages/views.rb:148]:
```suggestion
                trix_editor input: "content" autofocus: true
```

**Comment by mansakondo** [comment]
Ah, sorry I should have fixed the indentation myself

**Comment by joeldrapper** [comment]
No problem. Rubocop is really bad at fixing indentation because we use tabs instead of spaces.

**Pull request merged**